### PR TITLE
Clear cache processing

### PR DESCRIPTION
**Pull request by mjkozicki**
Add cached result processing.

### CODE REVIEW

File: www/cli/clearCache.php [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+$days_results_kept = null;
+if (GetSetting('days_results_kept')) {
+    $days_results_kept = GetSetting('days_results_kept');
+}
+
+$kept = 0;
+$deleted = 0;
+$log = fopen('./cli/clearedLocalResults.log', 'w');
+
+$MIN_DAYS = max($days_results_kept, 0.1);
```
Comment (claud-io) [inline www/cli/clearCache.php:36]: why 0.1? also we can just return early if days_results_kept is not set

File: www/cli/clearCache.php [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    }
+}
+
+/**
+* Check the given logfile for all matching tests
+*
+* @param mixed $logFile
+* @param mixed $match
+*/
+function CheckTest($testPath, $id, $elapsedDays)
```
Comment (claud-io) [inline www/cli/clearCache.php:115]:
```suggestion
function DeleteExpiredTest($testPath, $id, $elapsedDays)
```

File: www/cli/clearCache.php [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+Unlock($lock);
+
+/**
+* Recursively check within a given day
+*
+* @param mixed $dir
+* @param mixed $baseID
+*/
+function CheckDay($dir, $baseID, $elapsedDays)
```
Comment (claud-io) [inline www/cli/clearCache.php:83]: Those "checks"  are a bit misleading 


```suggestion
function DeleteTestsExpiredOn($dir, $baseID, $elapsedDays)
```

File: www/cli/clearCache.php (hunk 2 above)
Comment (mjkozicki) [reply www/cli/clearCache.php:115]: Changed name

**Review by claud-io — APPROVED** [review]

File: www/cli/clearCache.php [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+        if (is_numeric($year) && is_dir($yearDir) && $year != '.' && $year != '..') {
+            $months = scandir($yearDir);
+            foreach ($months as $month) {
+                $monthDir = "$yearDir/$month";
+                if (is_dir($monthDir) && $month != '.' && $month != '..') {
+                    $days = scandir($monthDir);
+                    foreach ($days as $day) {
+                        $dayDir = "$monthDir/$day";
+                        if (is_dir($dayDir) && $day != '.' && $day != '..') {
+                            $elapsedDays = ElapsedDays($year, $month, $day);
```
Comment (pmeenan) [inline www/cli/clearCache.php:54]: You can remove the $elapsedDays (and from the passed-into functions). That is the elapsed time based on the test ID but it's not used for this cache cleanup.  The actual cleanup is entirely done with "last accessed" time for the test.

File: www/cli/clearCache.php [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    if ($log) {
+        $logLine .= "\n";
+        fwrite($log, $logLine);
+    }
+}
+
+/**
+ * Calculate how many days have passed since the given day
+ */
+function ElapsedDays($year, $month, $day)
```
Comment (pmeenan) [inline www/cli/clearCache.php:166]: Function isn't needed.

**Review by pmeenan — APPROVED** [review]
LGTM, just a little opportunity for cleanup of unused code.

**Review by claud-io — APPROVED** [review]

**Pull request merged**